java/compaction/compaction-job-creation/src/main/java/sleeper/compaction/job/creation/CreateCompactionJobs.java

```diff
@@ -114,38 +117,56 @@ public enum Mode {
 
-    private void createJobsForTable(TableProperties tableProperties, StateStore stateStore) throws StateStoreException, IOException, ObjectFactoryException {
+    private Instant createJobsForTable(TableProperties tableProperties, StateStore stateStore) throws StateStoreException, IOException, ObjectFactoryException {
+        Instant startTime = Instant.now();
         TableStatus table = tableProperties.getStatus();
-        LOGGER.debug("Creating jobs for table {}", table);
+        LOGGER.info("Creating jobs for table {}", table);
 
+        Instant loadPartitionStartTime = Instant.now();
         List<Partition> allPartitions = stateStore.getAllPartitions();
 
         // NB We retrieve the information about all the active file references and filter
         // that, rather than making separate calls to the state store for reasons
         // of efficiency and to ensure consistency.
+        Instant loadFilesStartTime = Instant.now();
         List<FileReference> fileReferences = stateStore.getFileReferences();
 
+        Instant compactionStrategyCreationStartTime = Instant.now();
         CompactionStrategy compactionStrategy = objectFactory
                 .getObject(tableProperties.get(COMPACTION_STRATEGY_CLASS), CompactionStrategy.class);
         LOGGER.debug("Created compaction strategy of class {}", tableProperties.get(COMPACTION_STRATEGY_CLASS));
         CompactionJobFactory jobFactory = new CompactionJobFactory(instanceProperties, tableProperties, jobIdSupplier);
 
+        Instant indexCreationStartTime = Instant.now();
+        CompactionStrategyIndex index = new CompactionStrategyIndex(tableProperties.getStatus(), fileReferences, allPartitions);
+
+        Instant jobCreationStartTime = Instant.now();
         List<CompactionJob> compactionJobs = compactionStrategy.createCompactionJobs(
-                instanceProperties, tableProperties, jobFactory, fileReferences, allPartitions);
+                instanceProperties, tableProperties, jobFactory, index);
         LOGGER.info("Used {} to create {} compaction jobs for table {}", compactionStrategy.getClass().getSimpleName(), compactionJobs.size(), table);
 
+        Instant leftoverJobCreationStartTime = Instant.now();
+        Instant leftoverJobCreationFinishTime = leftoverJobCreationStartTime;
         if (mode == Mode.FORCE_ALL_FILES_AFTER_STRATEGY) {
             createJobsFromLeftoverFiles(tableProperties, jobFactory, fileReferences, allPartitions, compactionJobs);
+            leftoverJobCreationFinishTime = Instant.now();
         }
+
         int creationLimit = instanceProperties.getInt(CompactionProperty.COMPACTION_JOB_CREATION_LIMIT);
+        Instant limitJobsStartTime = leftoverJobCreationFinishTime;
+        Instant limitJobsFinishTime = limitJobsStartTime;
         if (compactionJobs.size() > creationLimit) {
             compactionJobs = reduceCompactionJobsDownToCreationLimit(compactionJobs, creationLimit);
+            limitJobsFinishTime = Instant.now();
```

Comment:
Minor nitpick, but these would be more consistent with the other times if you just had leftoverJobCreationStartTime, limitJobsStartTime and something like sendJobsStartTime. It doesn't seem necessary to default the times and replace them if the step runs, you can just sample the time between the steps whether they run or not.